app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/step/terminationdate/TerminationDateDestination.kt

```diff
@@ -62,7 +65,12 @@ internal fun TerminationDateDestination(
     uiState = uiState,
     windowSizeClass = windowSizeClass,
     dateValidator = viewModel.dateValidator,
-    submit = viewModel::submitSelectedDate,
+    submit = {
+      uiState.datePickerState.selectedDateMillis?.let {
+        val date = fromMilliseconds(it)
```

Comment:
As mentioned above, I think we can just inline this here, just as we had it inside `TerminationDateViewModel` before
```suggestion
        val date = Instant.fromEpochMilliseconds(selectedDateMillis).toLocalDateTime(TimeZone.UTC).date
```